api/src/main/java/com/xwiki/googleapps/internal/GoogleAppsXWikiObjects.java

```diff
@@ -0,0 +1,608 @@
+import org.xwiki.observation.ObservationManager;
+import org.xwiki.query.Query;
+import org.xwiki.query.QueryManager;
+import org.xwiki.stability.Unstable;
+
+import com.xpn.xwiki.XWikiContext;
+import com.xpn.xwiki.doc.XWikiAttachment;
+import com.xpn.xwiki.doc.XWikiDocument;
+import com.xpn.xwiki.objects.BaseObject;
+import com.xwiki.googleapps.DriveDocMetadata;
+import com.xwiki.googleapps.GoogleAppsException;
+
+class GoogleAppsXWikiObjects implements GoogleAppsConstants
+{
+    // environment
+    private final Logger log;
+
+    private final Provider<XWikiContext> contextProvider;
+
+    private final QueryManager queryManager;
+
+    private final DocumentReferenceResolver<String> documentResolver;
+
+    private final DocumentReferenceResolver<String> userResolver;
+
+    private final File permanentDir;
+
+    private boolean started;
+
+    private Boolean configActiveFlag;
+
+    private boolean useCookies;
+
+    private boolean skipLoginPage;
+
+    private boolean authWithCookies;
+
+    private String configAppName;
+
+    private String configClientId;
+
+    private String configClientSecret;
+
+    private String configDomain;
+
+    private boolean configScopeUseAvatar;
+
+    private boolean configScopeUseDrive;
+
+    private int configCookiesTTL;
+
+    // internal objects
+    private DocumentReference configDocRef;
+
+    private ObjectReference configObjRef;
+
+    private GoogleAppsEventListener eventListener;
+
+    private DocumentReference syncdocClassRef;
+
+    private DocumentReference gauthClassRef;
+
+    GoogleAppsXWikiObjects(Logger log, Provider<XWikiContext> contextProvider,
+            QueryManager queryManager, DocumentReferenceResolver<String> documentResolver,
+            DocumentReferenceResolver<String> userResolver, File permanentDir)
+    {
+        this.log = log;
+        this.contextProvider = contextProvider;
+        this.queryManager = queryManager;
+        this.documentResolver = documentResolver;
+        this.userResolver = userResolver;
+        this.permanentDir = permanentDir;
+    }
+
+    Boolean isActive()
+    {
+        return configActiveFlag;
+    }
+
+    boolean doesUseCookies()
+    {
+        return useCookies;
+    }
+
+    boolean doesSkipLoginPage()
+    {
+        return skipLoginPage;
+    }
+
+    boolean doesAuthWithCookies()
+    {
+        return authWithCookies;
+    }
+
+    String getConfigAppName()
+    {
+        return configAppName;
+    }
+
+    String getConfigClientId()
+    {
+        return configClientId;
+    }
+
+    String getConfigClientSecret()
+    {
+        return configClientSecret;
+    }
+
+    String getConfigDomain()
+    {
+        return configDomain;
+    }
+
+    boolean getConfigScopeUseAvatar()
+    {
+        return configScopeUseAvatar;
+    }
+
+    boolean doesConfigScopeUseDrive()
+    {
+        return configScopeUseDrive;
+    }
+
+    int getConfigCookiesTTL()
+    {
+        return configCookiesTTL;
+    }
```

Comment:
Why are these package protected?

Reply:
The complete class is package protected. My expectation that a public API is gathered in GoogleAppsManager. This configuration property is read from CookieAuthenticationPersistence.

I guess, however, that your are remarking that this method has no correspondance within the GoogleAppsScriptService and is, thus, not used in the views. That's right and I have removed this.